server/game/cards/02_SHD/units/GreySquadronYWing.ts

```diff
@@ -0,0 +1,26 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { RelativePlayer } from '../../../core/Constants';
+
+export default class GreySquadronYWing extends NonLeaderUnitCard {
+    protected override getImplementationId () {
+        return {
+            id: '9472541076',
+            internalName: 'grey-squadron-ywing',
+        };
+    }
+
+    public override setupCardAbilities () {
+        this.addOnAttackAbility({
+            title: 'An opponent chooses a unit or base they control. You may deal 2 damage to it',
+            targetResolver: {
+                controller: RelativePlayer.Opponent,
+                choosingPlayer: RelativePlayer.Opponent,
+                cardCondition: (card, context) => card.isUnit() || card.isBase(),
```

Comment:
Can we instead use `cardTypeFilter: [WildcardCardType.Unit, CardType.Base]` since that's a little simpler than having to run a handler